File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>
using namespace std;

bool solution(string s)
{
    stack<char> sta;
    
    sta.push(s[0]);
    for (int i = 1; i < s.size(); i++)
    {
        char newChar = s[i];
        if (newChar == '(')
        {
            sta.push(newChar);
        }
        
        if (newChar == ')')
        {
            if (sta.empty())
            {
                sta.push(newChar);
                continue;
            }
            
            if (sta.top() == '(')
            {
                sta.pop();
            }
            else
            {
                sta.push(newChar);
            }
        }
    }

    if (sta.size() == 0)
        return true;
    else
        return false;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    private Stack<char> sta = new Stack<char>();
    
    public bool solution(string s) 
    {
        foreach (char c in s)
        {
            if (c == '(')
            {
                sta.Push(c);
            }
            else
            {
                if (sta.Count() == 0)
                    return false;
                else
                    sta.Pop();
            }
        }
        
        if (sta.Count() == 0)
            return true;
        else
            return false;
    }
}